packages/proto-loader/package.json

```diff
@@ -36,20 +38,29 @@
   "files": [
     "LICENSE",
     "build/src/*.d.ts",
-    "build/src/*.js"
+    "build/src/*.js",
+    "build/bin/*.js"
   ],
+  "bin": {
+    "proto-loader-gen-types": "./build/bin/proto-loader-gen-types.js"
+  },
   "dependencies": {
+    "@types/long": "^4.0.1",
     "lodash.camelcase": "^4.3.0",
-    "protobufjs": "^6.8.6"
+    "long": "^4.0.0",
+    "protobufjs": "^6.10.0",
+    "yargs": "^16.1.1"
```

Comment:
What would be the benefit of making these changes?

Reply:
It would make it easier to manage dependencies for the users of this library.

If a user has version 15 of yargs defined in another package "*" would use that version instead of forcing 2 versions of yargs to exist, potentially causing conflicts.

You almost definitely don't want two versions of `protobufjs` to exist in `node_modules` since they would generate slightly different `.js` / `.d.ts` files. Using a peer dependency would force this dependency to use whatever package was already defined in the package. If you have a 'peer' dependency it doesn't automatically install when doing `yarn install` or the npm equivalent. Adding a `protobufjs` `dev` dependency alongside the `peer` will make the local development experience smoother.

In general it's nice to work with libraries that are permissive about the packages they depend on, unless they need something specific for a version.

Reply:
The problem is that I know that the version of yargs I am currently using works as I am using it, but I can't guarantee that every possible version of yargs that might ever be published will work the same way. And the NPM package structure should prevent version conflicts, so I don't see that as a relevant concern.

As for `protobufjs`, a peer dependency doesn't make sense. The purpose of a peer dependency is for when the user is expected to use that library directly, and the library with a peer dependency is a plugin or similar. But in this case, the proto-loader library is an abstraction layer over `protobufjs`, so it needs to use it directly, and it expects specific features from specific versions of `protobufjs`. I also don't see how different versions of `protobufjs` generating different output is a point in favor of making this change. Keeping the dependency as it is ensures that a specific version of `protobufjs` will be used when doing this code generation. Any other use of `protobufjs` outside of the context of this library is completely irrelevant to the function of this library.